pkg/commonvpc/vpc.go

```diff
@@ -353,18 +353,23 @@ func (nl *NaclLayer) AllowedConnectivity(src, dst vpcmodel.Node, isIngress bool)
 
 // RulesInConnectivity list of NACL rules contributing to the connectivity
 func (nl *NaclLayer) RulesInConnectivity(src, dst vpcmodel.Node,
-	conn *connection.Set, isIngress bool) (allowRes []vpcmodel.RulesInTable,
+	connQuery *connection.Set, isIngress bool) (allowRes []vpcmodel.RulesInTable,
 	denyRes []vpcmodel.RulesInTable, err error) {
 	for index, nacl := range nl.NaclList {
-		tableRelevant, allowRules, denyRules, err1 := nacl.rulesFilterInConnectivity(src, dst, conn, isIngress)
+		tableRelevant, allowRules, denyRules, err1 := nacl.rulesFilterInConnectivity(src, dst, connQuery, isIngress)
 		if err1 != nil {
 			return nil, nil, err1
 		}
 		if !tableRelevant {
 			continue
 		}
-		appendToRulesInFilter(&allowRes, &allowRules, index, true)
-		appendToRulesInFilter(&denyRes, &denyRules, index, false)
+		conn, err2 := nacl.AllowedConnectivity(src, dst, isIngress)
```

Comment:
Don't we also compute Allowed Connectivity somewhere else? Is this result reused?
Doubling the number of calls to `AllowedConnectivity` might double runtime, which might be bad for producing HTML.